### PR TITLE
Add configure option for MAC whitelist and blacklist.

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

File: src/core/Makefile.am [hunk 1]
```diff
@@ -89,6 +87,13 @@ SOURCES_COMMON                      = \
     utils/slaac_address.cpp           \
     $(NULL)
 
+if OPENTHREAD_ENABLE_MAC_WHITELIST
```
Comment (nibanks) [inline src/core/Makefile.am]: Can you add all these files to the Windows build and enable these features by default?

**Comment by mszczodrak** [comment]
Thanks @jwhui!

File: src/core/Makefile.am (hunk 1 above)
Comment (jwhui) [inline src/core/Makefile.am]: I believe these files should already be included in the windows build and the configure option is set to `1`.  What am I missing?

Comment (jwhui) [inline src/core/Makefile.am]: Ah yes, I updated this PR with the new header files.  Thanks.

**Pull request merged**